frontend/src/typeHelpers.ts

```diff
@@ -12,9 +12,11 @@ export type AnyObject = Record<string, unknown>;
  *
  * TODO: Implement the SDK & Patch logic -- this should stop being needed as things will be defined as Patches
  */
-export type RecursivePartial<T> = {
-  [P in keyof T]?: RecursivePartial<T[P]>;
-};
+export type RecursivePartial<T> = T extends object
+  ? {
+      [P in keyof T]?: RecursivePartial<T[P]>;
+    }
+  : T;
```

Comment:
I don't understand this change... why are you recursively making a non object partial?

Reply:
so i can include arrays

Reply:
actually never mind. i tried it without and its fine

Reply:
moved it back to orginal